client/lib/popup-monitor/index.js

```diff
@@ -98,6 +105,7 @@ PopupMonitor.prototype.checkStatus = function() {
 	if ( 0 === Object.keys( this.intervals ).length ) {
 		clearInterval( this.monitorInterval );
 		delete this.monitorInterval;
+		window.removeEventListener( 'message', this.onMessage );
```

Comment:
What happens when we have more than 1 open popup?

Reply:
It should work. Our listener should be receiving messages from all popups but then we filter those that come from our popup window here https://github.com/Automattic/wp-calypso/pull/25110/files/fd3591681f0ff9debfc069a6286f91eda69712f2#diff-1ea438880d24eea4297fcbcbf3fac3e5R22

Reply:
I meant that the first popup that closes will remove this listener for everyone, no?

Reply:
I don't think `PopupMonitor` is intended to be used for more than one popup, and even then if we have multiple instances of `PopupMonitor` it should work because `removeEventListener( event, listener )` only removes the given listener for that event not all listeners.

Reply:
aha, missed that each popup have it's own `PopupMonitor` instance 👍